v4/core/request_builder.go

```diff
@@ -76,6 +77,10 @@ type RequestBuilder struct {
 	// the "Content-Encoding" header will be added to the request with the
 	// value "gzip".
 	EnableGzipCompression bool
+
+	// RequestContext is an optional Context instance to be associated with the
+	// http.Request that is constructed by the Build() method.
+	ctx context.Context
```

Comment:
Do we want this field to be exportable/public ?

Reply:
Nope, I wanted to intentially "hide" it behind the "WithContext()" method